samples/modules/thrift/hello/doc/index.rst

```diff
@@ -0,0 +1,204 @@
+.. _thrift-hello-sample:
+
+Hello Sample Application
+########################
+
+Overview
+********
+
+This sample application includes a client a server implementing the RPC
+interface described in ``thrift/hello.thrift``. The purpose of this
+example is to demonstrate how components at different layer in thrift can
+be combined to build an application with desired features.
+
+
+Requirements
+************
+
+- :ref:`networking_with_host`
+- a Linux host with thrift installed
+- install Thrift dependencies for your host OS e.g. in Ubuntu
+
+.. code-block:: console
+   :caption: Install additional dependencies in Ubuntu
+
+   sudo apt install -y libboost-all-dev thrift-compiler libthrift-dev
+   pip install thrift
+
+Building and Running
+********************
+
+This application can be run on a Linux host, with either the server or the
+client in the QEMU environment, and the peer is built and run natively on
+the host.
+
+Building native client and server
+=================================
+
+.. code-block:: console
+
+   $ make -j -C client/
+   $ make -j -C server/
+
+Under ``client/``, 4 executables will be generated, and components
+used in each layer of them are listed below:
+
++----------------------+------------+--------------------+------------------+
+| hello_client         | TSocket    | TBufferedTransport | TBinaryProtocol  |
++----------------------+------------+--------------------+------------------+
+| hello_client_compact | TSocket    | TBufferedTransport | TCompactProtocol |
++----------------------+------------+--------------------+------------------+
+| hello_client_zlib    | TSocket    | TZlibTransport     | TBinaryProtocol  |
++----------------------+------------+--------------------+------------------+
+| hello_client_ssl     | TSSLSocket | TBufferedTransport | TBinaryProtocol  |
++----------------------+------------+--------------------+------------------+
+
+The same applies for the server. Only the client and the server with the
+same set of stacks can communicate.
+
+Additionally, there is a ``hello_client.py`` Python script that can be used
+interchangeably with the ``hello_client`` C++ application, simply to illustrate
+the cross-language capabilities of Thrift.
+
++----------------------+------------+--------------------+------------------+
+| hello_client.py      | TSocket    | TBufferedTransport | TBinaryProtocol  |
++----------------------+------------+--------------------+------------------+
+
+Build net-tools
+===============
+
+Fetch and build the Zephyr net-tools project, which is located in a separate
+Git repository:
+
+.. code-block:: console
+
+   $ git clone https://github.com/zephyrproject-rtos/net-tools
+   $ cd net-tools
+   $ make
```

Comment:
This is part of the default zephyr manifest, so if a user gets zephyr they should have it. And make? I've never had to run make on this repo on the test system I've used it for testing MCUmgr UDP things on - are you sure that's required?

Reply:
This is not a test. It's a sample. And running samples compiled for the host is really just an optional part of the sample, here for demonstration purposes only.

Several people have already complained very loudly about adding any kind of dependency to the GSG, even for tools that are *required* for the tests, so I will not be adding make as a requirement. In any case, I think make is probably already pulled in as a dependency, but certainly it's not much of a stretch for any competent developer.

Reply:
Not that make is a required tool or not, more that I've never needed to run make in the net-tools repo to use it

Reply:
This should be fine. We have many other samples doing the same thing.

Reply:
> Not that make is a required tool or not, more that I've never needed to run make in the net-tools repo to use it

Make is pulled in by `build-essential` in Ubuntu usually. It's used to build the vast majority of code that runs ... virtually everything.

It is referenced a number of times in our existing documentation, 2 of which are below.

https://docs.zephyrproject.org/latest/samples/net/sockets/echo/README.html#running-application-on-posix-host

https://docs.zephyrproject.org/latest/connectivity/networking/qemu_setup.html#prerequisites

Feel free to add an issue to "fix" make usage project-wide and impose cmake everywhere if that is your preference. It's probably outside the scope of this PR though.

Reply:
@cfriedt I don't think you are following @nordicjm 's point. He is not saying that having make installed is an issue. He is saying that having to run make in the net-tools repository seems like a new requirement.